### PR TITLE
ArchiSteamFarm: Add darwin as supported platforms

### PR DESCRIPTION
**Pull request by Endle**
## Description of changes

ArchiSteamFarm is a cross-platform software. I tested it on my mac, and current nix file works pretty well.

https://github.com/NixOS/nixpkgs/tree/c781bd7a53cf38594292e310a22e14a565d8a746/pkgs/applications/misc/ArchiSteamFarm

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@SuperSandro2000

### CODE REVIEW

File: pkgs/applications/misc/ArchiSteamFarm/default.nix [hunk 1]
```diff
@@ -76,7 +76,7 @@ buildDotnetModule rec {
     description = "Application with primary purpose of idling Steam cards from multiple accounts simultaneously";
     homepage = "https://github.com/JustArchiNET/ArchiSteamFarm";
     license = licenses.asl20;
-    platforms = [ "x86_64-linux" "aarch64-linux" ];
+    platforms = [ "x86_64-linux" "aarch64-linux" "x86_84-darwin" "aarch64-darwin" ];
```
Comment (wegank) [inline pkgs/applications/misc/ArchiSteamFarm/default.nix:79]:
```suggestion
    platforms = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
```

Comment (Endle) [reply pkgs/applications/misc/ArchiSteamFarm/default.nix]: Thank you for catching it. I've set it to platforms.unix

File: pkgs/applications/misc/ArchiSteamFarm/default.nix [hunk 2]
```diff
@@ -76,7 +76,7 @@ buildDotnetModule rec {
     description = "Application with primary purpose of idling Steam cards from multiple accounts simultaneously";
     homepage = "https://github.com/JustArchiNET/ArchiSteamFarm";
     license = licenses.asl20;
-    platforms = [ "x86_64-linux" "aarch64-linux" ];
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ArchiSteamFarm/default.nix:79]: I don't think Unix is correct here, we should take meta.platform from dotnet.

**Review by SuperSandro2000 — COMMENTED** [review]
Also note: I will be not using any of my time to fix it for Darwin. If it breaks, I'll mark it broken and someone needs to pick it up.

File: pkgs/applications/misc/ArchiSteamFarm/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/applications/misc/ArchiSteamFarm/default.nix:79]: Looks like the custom builder automatically defined it, so we can simply drop this line.

Comment (Endle) [reply pkgs/applications/misc/ArchiSteamFarm/default.nix:79]: > I don't think Unix is correct here, we should take meta.platform from dotnet.

With my understanding, https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/development/compilers/dotnet/default.nix shows that dotnet supports 
```
  runtimeIdentifierMap = {
    "x86_64-linux" = "linux-x64";
    "aarch64-linux" = "linux-arm64";
    "x86_64-darwin" = "osx-x64";
    "aarch64-darwin" = "osx-arm64";
    "x86_64-windows" = "win-x64";
    "i686-windows" = "win-x86";
  };
```


With my understanding, dotnet supports unix and windows(i686+x86_64). However, I haven't tested it on Windows. I'm not sure if I should add windows to ArchiSteamFarm's supported platforms

Comment (wegank) [reply pkgs/applications/misc/ArchiSteamFarm/default.nix:79]: I don't think there's the Windows support in nixpkgs, [the only sources](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/dotnet/versions/6.0.nix) being from the four platforms.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 264691` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**